src/shared_state.rs

```diff
@@ -123,19 +123,21 @@ impl Server {
 
         // wait for a truncated exponential backoff where it starts at 1ms and
         // doubles until it reaches 4 seconds;
-        let backoff_ms = if self.reconnects > 0 {
+        let backoff = if let Some(ref cb) = options.reconnect_delay_callback.0 {
+            (cb)(self.reconnects)
+        } else if self.reconnects > 0 {
             let log_2_four_seconds_in_ms = 12_u32;
             let truncated_exponent = std::cmp::min(
                 log_2_four_seconds_in_ms,
                 u32::try_from(std::cmp::min(u32::max_value() as usize, self.reconnects)).unwrap(),
             );
-            2_u64.checked_pow(truncated_exponent).unwrap()
+
+            let jitter = thread_rng().gen_range(0, 1000);
```

Comment:
This always fixed to 1ms? Might be reading it wrong.

Reply:
this is between 0 and 1000ms, which is the same as the Go client with up to 1s of default jitter